### PR TITLE
docs(contributing) rework

### PR DESCRIPTION
**Pull request by qdm12**
👋 Hey guys! I'm Quentin, I might join Chainsafe in a few days/weeks (waiting on papers), so I thought I would start contributing and reading up on stuff.

## Changes

Contributing document only

- Fix lint errors in document
- Update deadlinks
- Add syntax highlighting
- Fix indentation
- Remove typo / fix bad phrasing

## Tests

(Re-added to pass issue body CI check)

## Issues

I don't think an issue is necessary for this 😉

## Primary Reviewer 

*Feel free to edit my PR message to add a reviewer*

**Questions for reviewer(s)**:

1. `https://research.web3.foundation` seems **DOWN**. Maybe this is temporary, although it seems their DNS record is not there at all, pointing to a permanent *down*. Does anyone have a reliable link to the polkadot_re_spec.pdf document (Polkadot Runtime Specification)?

### CODE REVIEW

**Review by EclesioMeloJunior — APPROVED** [review]
lgtm!

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -4,96 +4,97 @@ Thank you for your interest in our implementation of the Polkadot Runtime Enviro
...
+    ```sh
+    git remote add gossamer https://github.com/ChainSafe/gossamer.git
+    git remote -v (you should see myrepo and gossamer in the list of remotes)
+    ```
 
-To start, check out our open issues. We recommend starting with an [issue labeled `Good First Issue`](https://github.com/ChainSafe/gossamer/issues?q=is%3Aopen+is%3Aissue+label%3A%22Good+First+Issue%22). Leave a comment to let us know that you would like to work on it. 
+5. **Find something to work on.**
 
-Another option is to improve gossamer where you see fit based on your evaluation of our code. In order to best facilitate collaboration, please create an issue before you start working on it.
+    To start, check out our open issues. We recommend starting with an [issue labeled `Good First Issue`](https://github.com/ChainSafe/gossamer/issues?q=is%3Aopen+is%3Aissue+label%3A%22Good+First+Issue%22). Leave a comment to let us know that you would like to work on.
```
Comment (noot) [inline .github/CONTRIBUTING.md:45]:
```suggestion
    To start, check out our open issues. We recommend starting with an [issue labeled `Good First Issue`](https://github.com/ChainSafe/gossamer/issues?q=is%3Aopen+is%3Aissue+label%3A%22Good+First+Issue%22). Leave a comment to let us know that you would like to work on it.
```

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -4,96 +4,97 @@ Thank you for your interest in our implementation of the Polkadot Runtime Enviro
...
 
-```
-$ git checkout feature-in-progress-branch
-```
+    Each time you work on the code be sure that you are working on the branch that you have created as opposed to your local copy of the gossamer repo. Keeping your changes segregated in this branch will make it easier to merge your changes into the repo later.
 
-**7. Test your changes.**
+    ```sh
+    git checkout feature-in-progress-branch
+    ```
```
Comment (noot) [inline .github/CONTRIBUTING.md:55]:
```suggestion
    ```sh
    git checkout -b feature-in-progress-branch
    ```
```

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -4,96 +4,97 @@ Thank you for your interest in our implementation of the Polkadot Runtime Enviro
...
+8. **Lint your changes.**
 
-```
-$ gometallinter ./...
-```
+    Before opening a pull request be sure to run the linter
+
+    ```sh
+    gometallinter ./...
+    ```
```
Comment (noot) [inline .github/CONTRIBUTING.md:85]: I think we use golangcilint now, so this should work:
```suggestion
    ```sh
    make lint
    ```
```

**Review by noot — COMMENTED** [review]
left a few comments, otherwise looks good!

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (qdm12) [reply .github/CONTRIBUTING.md:45]: Ah good 👀  sorry about that, I thought it was a `what` instead of `that` in that sentence 😅

**Comment by qdm12** [comment]
> left a few comments, otherwise looks good!

@noot yep I merged all your suggestions thanks for that 👍!

On another note, I might add things to this branch as I fiddle with this repo, so feel free to leave that PR a few more days (unless you have someone new coming soon). Thanks!

**Pull request merged**